src/tum_dlr_automl_for_eo/utils/custom_metrics.py

```diff
@@ -90,6 +90,48 @@ def perform_bils_algorithm(architectures, acc_list, starting_point, num_of_nei_t
     return found_arc_index, found_acc
 
 
+
+def get_hamming_distance(architecture_1, architecture_2):
+    return np.count_nonzero(architecture_1 != architecture_2)
+
+"""
+find starting points with high density of fully trained architectures, while simultaneously avoiding sampling twice
+from the same area 
+"""
+def find_k_starting_points(architectures, k,radius=3):
```

Comment:
So here we assume that architectures are already binary encoded right?

Reply:
Yes, that is the assumption